ledger/simulation/simulation_eval_test.go

```diff
@@ -848,6 +848,79 @@ int 0
 	})
 }
 
+func TestAppCallWithExtraBudget(t *testing.T) {
```

Comment:
Can you add an additional test where you exceed even the larger budget to make sure we still report that properly?

Reply:
see [extra testcase, that test cost budget exceed, even if with extra budget](https://github.com/algorand/go-algorand/pull/5354/commits/0b0eb970cdb2b151576338a116d0995162591d9e)